contracts/Blacksmith.sol

```diff
@@ -62,6 +63,7 @@ contract Blacksmith is Initializable, AccessControlUpgradeable {
     mapping(uint256 => address) currencies;
 
     mapping(uint256 => address) public links;
```

Comment:
Thoughts on trying the links mapping for saferandoms in both files? We can then avoid having the setSafeRandoms function and instead have a "setLink" with a link identifier constant, so we can set future references too without additional functions?

Reply:
That's a nice idea, implemented in both contracts